kafka-connect-ftp/src/test/scala/com/datamountaineer/streamreactor/connect/ftp/source/BySlicesTest.scala

```diff
@@ -169,4 +186,169 @@ class BySlicesTest extends AnyFunSuite with Matchers with BeforeAndAfter with St
+      allReadBytes.deep shouldBe fileContent1.deep
+
+      logger.info(s"===================================================")
+
+      //append content to file
+      val deltaContent = "extra".getBytes
+      server.putFile("/directory/file1.txt", fileContent1 ++ deltaContent)
+
+      val allReadBytes1: Array[Byte] = sftpPollUntilEnd(poller)
+
+      //Only the new delta
+      allReadBytes1.length shouldBe (fileContent1.size + deltaContent.size)
+    })
+  }
+
+  test("Sftp:Update mode by slices mode with MonitorTail and SimpleFileConverter :" +
+    " after update of file, only new data must be sent") {
+    withSftpServer(server => {
+      server.setPort(24)
+      server.addUser("demo", "password")
+
+      val offsets = new DummyOffsetStorage
+      val configMap = Map()
+        .updated(FtpSourceConfig.Address, "localhost:24")
+        .updated(FtpSourceConfig.protocol, "sftp")
+        .updated(FtpSourceConfig.User, "demo")
+        .updated(FtpSourceConfig.Password, "password")
+        .updated(FtpSourceConfig.RefreshRate, "PT0S")
+        .updated(FtpSourceConfig.MonitorTail, "/directory/:sftp_update_slice")
+        .updated(FtpSourceConfig.FileMaxAge, "PT952302H53M5.962S")
+        .updated(FtpSourceConfig.KeyStyle, "struct")
+        .updated(FtpSourceConfig.fileFilter, ".*")
+
+      val cfg = new FtpSourceConfig(configMap.asJava)
+
+      val poller: FtpSourcePoller = new FtpSourcePoller(cfg, offsets)
+
+      //push file
+      server.putFile("/directory/file1.txt", fileContent1)
+
+
+      val allReadBytes: Array[Byte] = sftpPollUntilEnd(poller)
+
+      allReadBytes.length shouldBe fileContent1.size
+      allReadBytes.deep shouldBe fileContent1.deep
+
+      logger.info(s"===================================================")
+
+      //append content to file
+      val deltaContent = "extra".getBytes
+      server.putFile("/directory/file1.txt", fileContent1 ++ deltaContent)
+
+      val allReadBytes1: Array[Byte] = sftpPollUntilEnd(poller)
+
+      //Only the new delta
+      allReadBytes1.length shouldBe deltaContent.size
+    })
+  }
+
+  def sftpPollUntilEnd(poller: FtpSourcePoller): Array[Byte] = {
+    var cnt = 0
+    logger.info("--------------------poll" + cnt + "-------------------------")
+    var slice = waitForSlice(poller)
+    var allReadBytes = new Array[Byte](0)
+    while (slice.lengthCompare(1) == 0) {
+      slice.head.topic shouldBe "sftp_update_slice"
+      val bytes = slice.head.value().asInstanceOf[Array[Byte]]
+      allReadBytes ++= bytes
+      logger.info(s"bytes.size=${bytes.length}")
+      cnt += 1
+      logger.info(s"--------------------poll$cnt-------------------------")
+      slice = waitForSlice(poller)
+    }
+    allReadBytes
+  }
+
+  private def waitForSlice(poller: FtpSourcePoller): Stream[SourceRecord] = {
+    var count = 0
+    var slice: Stream[SourceRecord] = poller.poll()
+    while (slice.isEmpty && count < 5) {
+      Thread.sleep(1000)
```

Comment:
Instead of sleeping in the test it would be better to use 
[eventually {}](http://doc.scalatest.org/1.8/org/scalatest/concurrent/Eventually.html)

Reply:
I don't know why but replacing the code by this one 

```
    eventually(timeout(Span(5, Seconds))) {
      slice = poller.poll()
      slice should not be empty
    }
```

it does not work.

If you have some time and can try it would be awesome

Reply:
I currently cannot run the test regardless of this change.

I am getting this:
`0 was not equal to 98901
ScalaTestFailureLocation: com.datamountaineer.streamreactor.connect.ftp.source.BySlicesTest at (BySlicesTest.scala:227)
Expected :98901
Actual   :0
`

I do believe your `eventually` change would work if the test would work.

Reply:
Is a Race condition.  With the Sleep the test has enough time to wait util the polling obtain the change and the test pass. 
Try first with the test without the refactor

Reply:
Sorry I wasn't clear.  I reverted the refactor and it is failing :disappointed: 

We are working on fixing github CI so that the tests run.  Then we will have more visibility.

Reply:
it's flaky test that like I said is a RC, but in my computer works fine, but of course is my computer. 
Once we can run in the pipeline let me know 😉

Reply:
ok - I will get back to you.